default.nix

```diff
@@ -0,0 +1,277 @@
+        the mail is stored.
+      '';
+    };
+
+    mail_dir = mkOption {
+      type = types.string;
+      default = "/var/vmail";
+      description = ''
+        Where to store the mail.
+      '';
+    };
+
+    certificate_scheme = mkOption {
+      type = types.enum [ 1 2 ];
+      default = 2;
+      description = ''
+        Certificate Files. There are three options for these.
+
+        1) You specify locations and manually copy certificates there.
+        2) You let the server create new (self signed) certificates on the fly.
+        3) You let the server create a certificate via `Let's Encrypt`. Note that
+           this implies that a stripped down webserver has to be started. This also
+           implies that the FQDN must be set as an `A` record to point to the IP of
+           the server. TODO: Explain more details
+
+        TODO: Only certificate scheme 1) and 2) work as of yet.
+      '';
+    };
+
+    cert_file = mkOption {
+      type = types.path;
+      example = "/root/mail-server.crt";
+      description = ''
+        Scheme 1)
+        Location of the certificate
+      '';
+    };
+
+    key_file = mkOption {
+      type = types.path;
+      example = "/root/mail-server.key";
+      description = ''
+        Scheme 1)
+        Location of the key file
+      '';
+    };
+
+    cert_dir = mkOption {
+      type = types.path;
+      default = "/var/certs";
+      description = ''
+        Sceme 2)
+        This is the folder where the certificate will be created. The name is
+        hardcoded to "cert-<domain>.pem" and "key-<domain>.pem" and the
+        certificate is valid for 10 years.
+      '';
+    };
+
+    enable_imap = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether to enable imap / pop3. Both variants are only supported in the
+        (sane) startTLS configuration. (TODO: Allow SSL ports). The ports are
+
+        110 - Pop3
+        143 - IMAP
+        587 - SMTP with login
+      '';
+    };
+
+    enable_pop3 = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to enable POP3. Both variants are only supported in the
+        (sane) startTLS configuration. (TODO: Allow SSL ports). The ports are
+
+        110 - Pop3
+        143 - IMAP
+        587 - SMTP with login
+      '';
+    };
+
+    # imapSsl = mkOption {} #< TODO
+    # pop3Ssl = mkOption {} #< TODO
+
+    virus_scanning = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        Whether to activate virus scanning. Note that virus scanning is _very_
+        expensive memory wise.
+      '';
+    };
+
+    dkim_signing = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Whether to activate dkim signing.
+        TODO: Explain how to put signature into domain record
+      '';
+    };
+
+    dkim_selector = mkOption {
+      type = types.string;
```

Comment:
I think, `types.str` should be used here
See comment
https://github.com/NixOS/nixpkgs/blob/4d545297d85c8f32f7ab496d0759f40d881bd61d/lib/types.nix#L145